include/nbl/asset/ICPUShader.h

```diff
@@ -29,35 +29,21 @@ class NBL_API ICPUShader : public IAsset, public IShader
 	protected:
 		virtual ~ICPUShader() = default;
 
-    private:
-        ICPUShader(core::smart_refctd_ptr<ICPUBuffer>&& _code, bool _isGLSL, const E_SHADER_STAGE stage, std::string&& filepathHint)
-			: IShader(stage, std::move(filepathHint)), m_code(std::move(_code)), m_containsGLSL(_isGLSL)
-		{}
-
 	public:
-        ICPUShader(
-			core::smart_refctd_ptr<ICPUBuffer>&& _spirv,
-			const E_SHADER_STAGE stage,
-			std::string&& filepathHint)
-			: ICPUShader(std::move(_spirv), false, stage, std::move(filepathHint))
-		{}
 
-        ICPUShader(
-			core::smart_refctd_ptr<ICPUBuffer>&& _glsl,
-			buffer_contains_glsl_t _buffer_contains_glsl,
-			const E_SHADER_STAGE stage,
-			std::string&& filepathHint)
-			: ICPUShader(std::move(_glsl), true, stage, std::move(filepathHint))
+		ICPUShader(core::smart_refctd_ptr<ICPUBuffer>&& code, const E_SHADER_STAGE stage, E_CONTENT_TYPE contentType, std::string&& filepathHint)
+			: IShader(stage, std::move(filepathHint)), m_code(std::move(code))
+			, m_contentType(contentType)
 		{}
 
 		ICPUShader(
-			const char* _glsl,
+			const char* code,
 			const E_SHADER_STAGE stage,
-			std::string&& filepathHint)
-			: ICPUShader(core::make_smart_refctd_ptr<ICPUBuffer>(strlen(_glsl) + 1u), true, 
-				stage, std::move(filepathHint))
+			E_CONTENT_TYPE contentType,
+			std::string&& filepathHint) 
+			: ICPUShader(core::make_smart_refctd_ptr<ICPUBuffer>(strlen(code) + 1u), stage, contentType, std::move(filepathHint))
```

Comment:
assert that content is not SPIR-V, otherwise strlen goes boom

Reply:
ok, but why?

Reply:
SPIR-V is not null terminated until a source string